File: mango-genome/frontend/src/types/blast.ts
export interface BlastHit {
  query_id: string;
  subject_id: string;
  pct_identity: number;
  alignment_length: number;
  mismatches: number;
  gap_openings: number;
  query_start: number;
  query_end: number;
  subject_start: number;
  subject_end: number;
  evalue: number;
  bit_score: number;
}

export interface BlastPlots {
  identity_plot: string;   // base64 PNG
  evalue_plot: string;
  bitscore_plot: string;
}

export interface BlastResponse {
  query_length: number;
  gc_content: number;
  hits: BlastHit[];
  total_hits: number;
  plots: BlastPlots | null;
  csv_data: string;
}

export interface DatabaseList {
  nucleotide: string[];
  protein: string[];
}

File: mango-genome/backend/schemas/blast.py
from typing import Literal, List, Optional
from pydantic import BaseModel


class BlastHit(BaseModel):
    query_id: str
    subject_id: str
    pct_identity: float
    alignment_length: int
    mismatches: int
    gap_openings: int
    query_start: int
    query_end: int
    subject_start: int
    subject_end: int
    evalue: float
    bit_score: float


class BlastPlots(BaseModel):
    identity_plot: str   # base64 PNG
    evalue_plot: str     # base64 PNG
    bitscore_plot: str   # base64 PNG


class BlastResponse(BaseModel):
    query_length: int
    gc_content: float
    hits: List[BlastHit]
    total_hits: int
    plots: Optional[BlastPlots] = None
    csv_data: str


class DatabaseListResponse(BaseModel):
    nucleotide: List[str]
    protein: List[str]
